Add comprehensive Keycloak integration documentation

michaelstingl opened the pull request:
This PR adds detailed documentation for the Keycloak integration with OpenCloud, replacing the empty draft page with comprehensive content.

## Content added:

- Overview of OpenCloud's authentication strategy
- Step-by-step setup instructions with Docker Compose
- OpenCloud configuration guide
- Detailed client configuration for all client types:
  - Web client
  - Desktop client
  - Mobile apps (Android/iOS)
  - Additional clients (Cyberduck)
- Advanced configuration (Backchannel Logout)
- Troubleshooting advice
- User and group management overview

## Related Issues and PRs
- Addresses #97 - "Document how to Integrate Keycloak / external Idp"
- Complements opencloud-eu/opencloud#589 - "Harmonize Keycloak Client Configurations with Built-in IdP"

micbar commented on docs/admin/30-configuration/30-keycloak.md:
````diff
@@ -1,6 +1,161 @@
+   ```
+4. Start the deployment with Keycloak:
+   ```bash
+   docker-compose -f docker-compose.yml -f keycloak.yml up -d
````
```suggestion
   docker-compose up -d
```

micbar replied:
We do not need to provide the file paths. The magic lies in the last line of the `.env` file 🪄

michaelstingl commented:
Thank you @micbar for pointing this out! You're absolutely right about simplifying the Docker Compose command.

I've implemented the suggested change and added explanatory notes about how this "magic mechanism" works with the COMPOSE_FILE variable in the .env file. The changes are now included in the latest commit.

Thanks for the improvement! 👍

micbar commented:
@michaelstingl please rebase to get a green CI run.

michaelstingl commented:
Rebased onto current main branch. All requested changes have been implemented.
